src/utils/src/arg_parser.rs

```diff
@@ -0,0 +1,567 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+use std::env;
+use std::fmt;
+use std::result;
+
+pub type Result<T> = result::Result<T, Error>;
+
+const ARG_SEPARATOR: &str = "--";
```

Comment:
The `--` string is used both as a separator (e.g. `arg1 arg2 -- forwarded_arg1`), and as a prefix (e.g. `--arg`). We should make that distinction, and avoid code like `starts_with(ARG_SEPARATOR)` which sounds confusing.